WordPress/src/main/java/org/wordpress/android/util/UploadWorkRequest.kt

```diff
@@ -0,0 +1,67 @@
+package org.wordpress.android.util
+
+import android.content.Context
+import androidx.work.Constraints
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.ExistingWorkPolicy
+import androidx.work.NetworkType
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.Operation
+import androidx.work.PeriodicWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.WorkRequest
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import androidx.work.workDataOf
+import org.wordpress.android.WordPress
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.store.SiteStore
+import org.wordpress.android.ui.uploads.LocalDraftUploadStarter
+import java.util.concurrent.TimeUnit.HOURS
+import javax.inject.Inject
+
+class AutoUploadWorker(
+    appContext: Context,
+    workerParams: WorkerParameters
+) : Worker(appContext, workerParams) {
+    @Inject lateinit var localDraftUploadStarter: LocalDraftUploadStarter
+    @Inject lateinit var siteStore: SiteStore
+
+    override fun doWork(): Result {
+        (applicationContext as WordPress).component().inject(this)
+        when (val localSiteId = inputData.getInt(WordPress.LOCAL_SITE_ID, -1)) {
+            -1 -> localDraftUploadStarter.queueUploadFromAllSites()
+            else -> siteStore.getSiteByLocalId(localSiteId)?.let { localDraftUploadStarter.queueUploadFromSite(it) }
+        }
+        return Result.success()
```

Comment:
I'm not sure how we can (and if we can) make sure the service has started and is running. I'm also not sure about the app lifecycle in this case, I guess it's killable after doWork() has finished.

Reply:
> I think the best solution would be to bypass the UploadService in that specific case and have the underlying upload code ran in this doWork() functions.

You're right. I didn't think about that. It will probably be killed. This may require a significant refactor of `UploadService`. 🤔

Reply:
>I guess it's killable after doWork() has finished.

Yes, I believe so.

>I think the best solution would be to bypass the UploadService, extract upload functions out of the service and run that in the doWork() function. The problem is that we have a lot of logic in our UploadService and we can't just reuse the upload/callback functions.

I agree, that'd be the best solution, but it requires quite a huge refactoring.


Since, the task itself isn't vital - nothing really happens if we upload the posts/pages later. I'd suggest just making sure the service is at least promoted to a foreground service. I'm haven't checked why, but when we start it from the EditPostActivity, it gets promoted to a foreground service, but when it's started from the `LocalDraftUploadStarter` it doesn't get promoted.

Reply:
I agree it's not vital and I don't think we should refactor the UploadService for this. We should keep this in mind if we ever refactor it.

Reply:
wrapped in a `runBlocking` block in aaaa0b0